app/javascript/react/components/RestaurantShow.js

```diff
@@ -1,33 +1,102 @@
-import React, { useState, useEffect } from 'react'
-
-const RestaurantShow = (props) => {
-    const [restaurant, setRestaurant] = useState({})
-
-    const id = props.match.params.id
-
-    useEffect(() => {
-      fetch(`/api/v1/restaurants/${id}`)
-      .then(response => {
-        if (response.ok) {
-          return response
-        } else {
-          let errorMessage = `${response.status} (${response.statusText})`,
-            error = new Error(errorMessage);
-          throw(error);
-        }
-      })
-      .then(response => response.json())
-      .then(responseBody => {
-        setRestaurant(responseBody)
-      })
-    .catch(error => console.error(`Error in fetch: ${error.message}`))
-    }, [])
+import React, { useState, useEffect } from "react";
+import ReviewForm from "./ReviewForm";
+import ReviewTile from './ReviewTile';
+
+const RestaurantShow = props => {
+  const [restaurant, setRestaurant] = useState({});
+  const [getReviewData, setReviewData] = useState([]);
+  const [getNewReview, setNewReview] = useState({
+    masks_customers: false, 
+    masks_employees: false, 
+    social_distancing: false, 
+    party_size: false
+  });
+
+  const id = props.match.params.id;
+
+  const handleSubmit = event => { 
+    
+    event.preventDefault();
+
+    const formPayload = {
+      review: getNewReview
+    }
+
+    fetch(`/api/v1/restaurants/${restaurant.id}/reviews`, {
+      credentials: "same-origin",
+      method: "POST",
+      body: JSON.stringify(formPayload),
+      headers: {
+        'Accept': 'application/json',
+        "Content-Type": "application/json"
+      }
+  })
+  .then(response => {
+    if (response.ok) {
+      return response;
+  } else {
+      const errorMessage = `${response.status} (${response.statusText})`;
+      const error = new Error(errorMessage);
+      throw error;
+    }
+  })
+  .then(response => response.json())
+  .then(body => {console.log(body); setReviewData([...getReviewData, body]);})
+  .catch(error => console.error(`Error in fetch: ${error.message}`))
+  }
+
+  const handleInputChange = event => {
+
+    const target = event.currentTarget;
+    const value = target.type === 'checkbox' ? target.checked : Number(target.value);
+
+    setNewReview({
+      ...getNewReview,
+      [event.currentTarget.name]: value
+    });
+  }
+
+  useEffect(() => {
+    fetch(`/api/v1/restaurants/${id}`)
+    .then(response => {
+      if (response.ok) {
+        return response
+      } else {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw(error);
+      }
+    })
+    .then(response => response.json())
+    .then(responseBody => setRestaurant(responseBody))
+  .catch(error => console.error(`Error in fetch: ${error.message}`))
+  }, [])
+
+  const reviewTileArray = () => restaurant.reviews.map((review) => (
```

Comment:
this doesn't need to be a function
```
const reviewTileArray = restaurant.reviews.map( ...
```

Reply:
this was fixed in salims pull request 16 edits